web/src/app/admin/connectors/gitlab/page.tsx

```diff
@@ -0,0 +1,250 @@
+import { Card, Divider, Text, Title } from "@tremor/react";
+import { AdminPageTitle } from "@/components/admin/Title";
+
+const Main = () => {
+  const { mutate } = useSWRConfig();
+  const {
+    data: connectorIndexingStatuses,
+    isLoading: isConnectorIndexingStatusesLoading,
+    error: isConnectorIndexingStatusesError,
+  } = useSWR<ConnectorIndexingStatus<any, any>[]>(
+    "/api/manage/admin/connector/indexing-status",
+    fetcher
+  );
+
+  const {
+    data: credentialsData,
+    isLoading: isCredentialsLoading,
+    error: isCredentialsError,
+    refreshCredentials,
+  } = usePublicCredentials();
+
+  if (
+    (!connectorIndexingStatuses && isConnectorIndexingStatusesLoading) ||
+    (!credentialsData && isCredentialsLoading)
+  ) {
+    return <LoadingAnimation text="Loading" />;
+  }
+
+  if (isConnectorIndexingStatusesError || !connectorIndexingStatuses) {
+    return <div>Failed to load connectors</div>;
+  }
+
+  if (isCredentialsError || !credentialsData) {
+    return <div>Failed to load credentials</div>;
+  }
+
+  const gitlabConnectorIndexingStatuses: ConnectorIndexingStatus<
+    GitlabConfig,
+    GitlabCredentialJson
+  >[] = connectorIndexingStatuses.filter(
+    (connectorIndexingStatus) =>
+      connectorIndexingStatus.connector.source === "gitlab"
+  );
+  const gitlabCredential: Credential<GitlabCredentialJson> | undefined =
+    credentialsData.find(
+      (credential) => credential.credential_json?.gitlab_access_token
+    );
+
+  return (
+    <>
+      <Title className="mb-2 mt-6 ml-auto mr-auto">
+        Step 1: Provide your access token
+      </Title>
+      {gitlabCredential ? (
+        <>
+          {" "}
+          <div className="flex mb-1 text-sm">
+            <p className="my-auto">Existing Access Token: </p>
+            <p className="ml-1 italic my-auto">
+              {gitlabCredential.credential_json.gitlab_access_token}
+            </p>{" "}
+            <button
+              className="ml-1 hover:bg-hover rounded p-1"
+              onClick={async () => {
+                await adminDeleteCredential(gitlabCredential.id);
+                refreshCredentials();
+              }}
+            >
+              <TrashIcon />
+            </button>
+          </div>
+        </>
+      ) : (
+        <>
+          <Text>
+            If you don&apos;t have an access token, read the guide{" "}
+            <a
+              className="text-blue-500"
+              href="https://docs.danswer.dev/connectors/gitlab"
+              target="_blank"
+            >
+              here
+            </a>{" "}
+            on how to get one from Gitlab.
+          </Text>
+          <Card className="mt-4">
+            <CredentialForm<GitlabCredentialJson>
+              formBody={
+                <>
+                  <TextFormField
+                    name="gitlab_url"
+                    label="GitLab URL:"
+                    type="text"
+                    placeholder="https://gitlab.com"
```

Comment:
Can we give a bit more description after this like:
"If you are using GitLab Cloud, keep the default value below"

Reply:
added <Text/>